### DISCUSSION TITLE
Is there any way I can use integer input in place of a string input?

### DISCUSSION THREAD

**Discussion by armahapa** [reaktoro/reaktoro#384] [Q&A]
### Description

Let's take the following section from the example (examples/cpp/ex-kinetics-calcite-using-embedded-palandri-kharaka-params.rkp.cpp)
    
    SupcrtDatabase db("supcrtbl");

    ChemicalSystem system(db,
        AqueousPhase("H2O(aq) H+ OH- Ca+2 HCO3- CO3-2 CO2(aq)").set(ActivityModelDavies()),
        MineralPhase("Calcite"),
        MineralReaction("Calcite").setRateModel(ReactionRateModelPalandriKharaka(params)),
        MineralSurface("Calcite", 5.0, "cm2", 70, "mg", 0.667)  // surface area = (initial surface area)*((current mass)/(initial mass))**power = (5 cm2)*((current mass in mg)/(70 mg))**0.667
    );
    
Here species inputs like H+, OH- etc have data type of string. Is there a way to perform the same without passing the string?
 I was wondering if there exists a dictionary that has integer index numbers and if we can use the index for this type of module.

### What is your operating system?

Windows

**Comment by allanleal** [comment]
Hi @armahapa - I did not fully understand your question. A mock example demonstrating what you'd like to do would be helpful. But I think the answer would still most likely no - for the initialization code above, to define a chemical system, you use string for names of species, not integers.
